src/Text/CSL/Parser.hs:66:40: No instance for (Default X.ParseSettings)

Hello,

When compiling pandoc-citeproc 0.14.3.1 on OpenBSD 6.3 ghc version 8.2.2, cabal version 2.0.0.1 I get the following error:

```
[19 of 23] Compiling Text.CSL.Parser  ( src/Text/CSL/Parser.hs, dist/build/Text/CSL/Parser.o )

src/Text/CSL/Parser.hs:66:40: error:
    * No instance for (Default X.ParseSettings)
        arising from a use of `def'
    * In the first argument of `X.parseLBS_', namely `def'
      In the second argument of `($)', namely `X.parseLBS_ def f'
      In the expression: fromDocument $ X.parseLBS_ def f
   |
66 |   let cur = fromDocument $ X.parseLBS_ def f
   |                                        ^^^

src/Text/CSL/Parser.hs:83:57: error:
    * No instance for (Default X.ParseSettings)
        arising from a use of `def'
    * In the first argument of `X.parseLBS_', namely `def'
      In the second argument of `(.)', namely `X.parseLBS_ def'
      In the second argument of `(.)', namely
        `fromDocument . X.parseLBS_ def'
   |
83 | parseCSL' = parseCSLCursor . fromDocument . X.parseLBS_ def
   |                                                         ^^^
cabal: Leaving directory '.'
cabal: Error: some packages failed to install:
pandoc-citeproc-0.14.3.1-Dr7I7A8ZT6wBZCFp96VdKN failed during the building
phase. The exception was:
ExitFailure 1
```

I can reproduce (a similar?) error in `cabal exec ghci` by importing both `Data.Default` and `Text.XML`:
```

citrin$ cabal exec ghci
GHCi, version 8.2.2: http://www.haskell.org/ghc/  :? for help
Prelude> import Data.Default
Prelude Data.Default> import Text.XML
Prelude Data.Default Text.XML> parseLBS_ def

<interactive>:3:11: error:
    Ambiguous occurrence `def'
    It could refer to either `Text.XML.def',
                             imported from `Text.XML'
                             (and originally defined in `Data.Default.Class')
                          or `Data.Default.def', imported from `Data.Default'
```

So I've gone ahead and explicitly used `X.def` in Parser.hs and that seems to fix the issue.  Will next issue a PR.

Thanks,
Ed

--- a/src/Text/CSL/Parser.hs
+++ b/src/Text/CSL/Parser.hs
@@ -21,7 +21,6 @@ import Prelude
 import qualified Control.Exception      as E
 import           Control.Monad          (when)
 import qualified Data.ByteString.Lazy   as L
-import           Data.Default
 import           Data.Either            (lefts, rights)
 import qualified Data.Map               as M
 import           Data.Maybe             (fromMaybe, listToMaybe)
@@ -45,7 +44,7 @@ parseCSL = parseCSL' . fromStringLazy
 -- | Parse locale.  Raises 'CSLLocaleException' on error.
 parseLocale :: String -> IO Locale
 parseLocale locale =
-  parseLocaleElement . fromDocument . X.parseLBS_ def <$> getLocale locale
+  parseLocaleElement . fromDocument . X.parseLBS_ X.def <$> getLocale locale
 
 -- | Merge locale into a CSL style.
 localizeCSL :: Maybe String -> Style -> IO Style
@@ -63,7 +62,7 @@ readCSLFile mbLocale src = do
   f <- case fetchRes of
             Left err         -> E.throwIO err
             Right (rawbs, _) -> return $ L.fromChunks [rawbs]
-  let cur = fromDocument $ X.parseLBS_ def f
+  let cur = fromDocument $ X.parseLBS_ X.def f
   -- see if it's a dependent style, and if so, try to fetch its parent:
   let pickParentCur = get "link" >=> attributeIs (X.Name "rel" Nothing Nothing) "independent-parent"
   let parentCur = cur $/ get "info" &/ pickParentCur
@@ -80,7 +79,7 @@ readCSLFile mbLocale src = do
            readCSLFile mbLocale' y
 
 parseCSL' :: L.ByteString -> Style
-parseCSL' = parseCSLCursor . fromDocument . X.parseLBS_ def
+parseCSL' = parseCSLCursor . fromDocument . X.parseLBS_ X.def
 
 parseCSLCursor :: Cursor -> Style
 parseCSLCursor cur =

